include/swift/AST/DiagnosticsSema.def

```diff
@@ -4807,6 +4807,12 @@ ERROR(return_init_non_nil,none,
       "'nil' is the only return value permitted in an initializer",
       ())
 
+ERROR(implicit_throws_super_init,none,
+      "missing call to superclass's initializer; "
```

Comment:
Hmm... I wonder if developers would find the second clause more coherent if the first one mentioned that the call cannot be synthesized. For example: 

`call to superclass's initializer is missing and cannot be synthesized; 'super.init' is a throwing initializer and requires either an explicit call or that this initializer is also marked as 'throws'`.

vs.

`missing call to superclass's initializer; 'super.init' is a throwing initializer and requires either an explicit call or that this initializer is also marked as 'throws'`

Longer, yes, but I’m not a fan of this "brevity beats clarity" policy we appear to be expected to follow for Swift diagnostics.

Reply:
cc @hborla

Reply:
I agree the first one makes it clearer but will wait a bit to see if someone with more experience than me has something to say about the "brevity beats clarity" tradeoff.

Reply:
> I believe we would have to change this on the async version of them too.

Yeah, fair enough. Better save this for another PR.